File: 1217-relative-sort-array/1217-relative-sort-array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        vector<int> freq(1001, 0);

        for(int x : arr1){
            freq[x]++;
        }

        vector<int> res;

        for(int x : arr2){
            while(freq[x] > 0){
                res.push_back(x);
                freq[x]--;
            }
        }

        for(int x = 0; x < 1001; x++){
            while(freq[x] > 0){
                res.push_back(x);
                freq[x]--;
            }
        }
        return res;
    }
};

File: 1217-relative-sort-array/1217-relative-sort-array.py
class Solution(object):
    def relativeSortArray(self, arr1, arr2):
        """
        :type arr1: List[int]
        :type arr2: List[int]
        :rtype: List[int]
        """
        freq = [0] * 1001

        for i in arr1:
            freq[i] += 1

        res = []

        for i in arr2:
            while(freq[i] > 0):
                res.append(i)
                freq[i] -= 1

        for i in range(1001):
            while(freq[i] > 0):
                res.append(i)
                freq[i] -= 1

        return res